editor.planx.uk/src/@planx/components/MultipleFileUpload/Modal.tsx

```diff
@@ -60,3 +72,78 @@ export const FileTaggingModal = (props: FileTaggingModalProps) => {
     </Dialog>
   );
 };
+
+interface SelectMultipleProps extends SelectProps {
+  name: string;
+  fileList: FileList;
+}
+
+const SelectMultiple = (props: SelectMultipleProps) => {
+  const { name, fileList } = props;
+  const [tags, setTags] = React.useState<string[]>([]);
+
+  const handleChange = (event: SelectChangeEvent<typeof tags>) => {
+    const {
+      target: { value },
+    } = event;
+    setTags(
+      // On autofill we get a stringified value.
+      typeof value === "string" ? value.split(",") : value
+    );
```

Comment:
This is actually a nice setup - we get a discrete change event here each time a checkbox is checked / unchecked. 

Maybe we simplify things a bit and we can just assign slots -> fileList directly here and not map a list of tags on modal submit / close?

So something like...
 - When we check a box...
 - We know the slot, we know the tag, we know the category (required/recommended/optional)
 - Assign the slot to the filetype, call `setFileState()`

Maybe something like...
 
 ```ts
const updatedFileList = {...fileList};
updatedFileList[category].find(fileType => fileType.name === name).slot = slot;
setFileList(updatedFileList);
 ```
 
 I guess unchecking would be just the same but `slot === undefined`